src/lib/features/project/project-service.ts

```diff
@@ -1282,21 +1292,18 @@ export default class ProjectService {
     async statusJob(): Promise<void> {
         const projects = await this.projectStore.getAll();
 
-        const statusUpdates = await Promise.all(
-            projects.map((project) => this.getStatusUpdates(project.id)),
-        );
-
-        await Promise.all(
-            statusUpdates.map((statusUpdate) => {
-                return this.projectStatsStore.updateProjectStats(
-                    statusUpdate.projectId,
-                    statusUpdate.updates,
-                );
-            }),
-        );
+        // run one project status update at a time every
+        void batchExecute(projects, 1, 30_000, async (project) => {
+            const statusUpdate = await this.getStatusUpdates(project.id);
+            await this.projectStatsStore.updateProjectStats(
+                statusUpdate.projectId,
+                statusUpdate.updates,
+            );
+        });
     }
 
     async getStatusUpdates(projectId: string): Promise<ICalculateStatus> {
+        const stopTimer = this.timer('getStatusUpdates');
```

Comment:
because we don't have a total scheduled job timer working due to async batching we need another timer on the most expensive function itself